### ISSUE TITLE
Getting error while converting to Google Colab TPU

### ISSUE THREAD

**Issue by Sgmoid** [matterport/Mask_RCNN#1036]
This is the code

```
TPU_WORKER = 'grpc://' + os.environ['COLAB_TPU_ADDR']

import tensorflow as tf
tpu_model = tf.contrib.tpu.keras_to_tpu_model(
model.keras_model,
strategy=tf.contrib.tpu.TPUDistributionStrategy(
tf.contrib.cluster_resolver.TPUClusterResolver(TPU_WORKER)))

```
Which is giving below error

ValueError: 
Layer <keras.engine.topology.InputLayer object at 0x7f58574f1940> has a 
variable shape in a non-batch dimension.  TPU models must
have constant shapes for all operations.

You may have to specify `input_length` for RNN/TimeDistributed layers.

Layer: <keras.engine.topology.InputLayer object at 0x7f58574f1940>
Input shape: (None, None, None, 3)
Output shape: (None, None, None, 3)

**Comment by bishesh16** [comment]
+1 @Sgmoid Did you manage to solve this issue ?

**Comment by Sgmoid** [comment]
> +1 @Sgmoid Did you manage to solve this issue ?

No. We have moved to AWS GPU. Google Colab GPUs are slow. We wanted to use Google Colab TPUs. But we are running into above issue.

**Comment by h-ann** [comment]
Hi @Sgmoid ,
Could you please advise, which Instance type on aws you chose?
